hoodie-client/src/main/java/com/uber/hoodie/HoodieWriteClient.java

```diff
@@ -937,141 +946,111 @@ public String startCommit() {
   }
 
-  /**
-   * Provides a new commit time for a compaction (commit) operation
-   */
-  public String startCompaction() {
-    String commitTime = HoodieActiveTimeline.createNewCommitTime();
-    logger.info("Generate a new commit time " + commitTime);
-    startCompactionWithTime(commitTime);
-    return commitTime;
-  }
 
   /**
-   * Since MOR tableType default to {@link HoodieTimeline#DELTA_COMMIT_ACTION}, we need to
-   * explicitly set to {@link HoodieTimeline#COMMIT_ACTION} for compaction
+   * Schedules a new compaction instant
+   * @param extraMetadata
+   * @return
    */
-  public void startCompactionWithTime(String commitTime) {
-    HoodieTable<T> table = HoodieTable.getHoodieTable(
-        new HoodieTableMetaClient(jsc.hadoopConfiguration(), config.getBasePath(), true), config);
-    HoodieActiveTimeline activeTimeline = table.getActiveTimeline();
-    String commitActionType = HoodieTimeline.COMMIT_ACTION;
-    activeTimeline.createInflight(new HoodieInstant(true, commitActionType, commitTime));
+  public String scheduleCompaction(Optional<Map<String, String>> extraMetadata) throws IOException {
+    String instantTime = HoodieActiveTimeline.createNewCommitTime();
+    logger.info("Generate a new instant time " + instantTime);
+    scheduleCompactionWithTime(instantTime, extraMetadata);
+    return instantTime;
   }
 
   /**
-   * Performs a compaction operation on a dataset. WARNING: Compaction operation cannot be executed
-   * asynchronously. Please always use this serially before or after an insert/upsert action.
+   * Schedules a new compaction instant with passed-in instant time
+   * @param instantTime     Compaction Instant Time
+   * @param extraMetadata   Extra Metadata to be stored
    */
-  public JavaRDD<WriteStatus> compact(String commitTime) throws IOException {
-    // Create a Hoodie table which encapsulated the commits and files visible
-    HoodieTable<T> table = HoodieTable.getHoodieTable(
-        new HoodieTableMetaClient(jsc.hadoopConfiguration(), config.getBasePath(), true), config);
-    // TODO : Fix table.getActionType for MOR table type to return different actions based on delta or compaction
-    writeContext = metrics.getCommitCtx();
-    JavaRDD<WriteStatus> statuses = table.compact(jsc, commitTime);
-    // Trigger the insert and collect statuses
-    statuses = statuses.persist(config.getWriteStatusStorageLevel());
-    String actionType = HoodieActiveTimeline.COMMIT_ACTION;
-    commitOnAutoCommit(commitTime, statuses, actionType);
-    return statuses;
+  public void scheduleCompactionWithTime(String instantTime, Optional<Map<String, String>> extraMetadata)
+      throws IOException {
+    HoodieTableMetaClient metaClient = new HoodieTableMetaClient(jsc.hadoopConfiguration(),
+        config.getBasePath(), true);
+    // if there are inflight ingestions, their instantTime must not be less than that of compaction instant time
```

Comment:
In our current deployment model there are no ingestion jobs running in parallel since only the ingestion job schedules compaction at the moment, correct ?

Reply:
Correct.